### PR TITLE
docs(PPDSC-2388): link-documentation

### PR DESCRIPTION
**Pull request by RashikaNewsUK**
PPDSC-2388

**What**

1. Link documentation UI.

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [x] Updated relevant documentation

**I have tested manually:**
- [x] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [x] The screen reader reads and flows through the elements as expected.
- [x] There are no new errors in the browser console coming from this PR.
- [x] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [x] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2388-link-documentation
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2388-link-documentation/storybook

File: site/pages/components/link.tsx [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+          </Link>{' '}
+          has additional requirements for body text links that are not
+          underlined by default.
+        </>
+      ),
+    }}
+    componentAPI={{
+      components: [
+        {
+          title: 'Link',
```
Comment (mutebg) [inline site/pages/components/link.tsx:347]: IMO we should have 2 separate components, LinkInline and LinkStandalone

File: site/pages/components/link.tsx [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+            },
+          ],
+          overridesRows: [
+            {
+              attribute: 'inlineLink.stylePreset',
+              type: 'MQ<string>',
+              default: 'linkInline',
+              description:
+                'If provided, overrides the stylePresets of the inlineLink.',
+            },
```
Comment (mutebg) [inline site/pages/components/link.tsx:423]: All overrides should be without `inlineLink` and `standaloneLink` and again we need to have 2 tables

File: site/pages/components/link.tsx [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+              default: 'medium',
+              description: 'Defines the size of the link.',
+            },
+            {
+              name: 'visited',
+              type: 'boolean',
+              default: 'false',
+              description:
+                'If true, renders the undefined link in a visited state.',
+            },
```
[the remark this answers is not in the document]
Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:385]: Sure @mutebg , Let me check. -thanks!

File: site/pages/components/link.tsx [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+                </>
+              ),
+              required: true,
+            },
+            {
+              name: 'size',
+              type: ['small', 'medium', 'large'],
+              default: 'medium',
+              description: 'Defines the size of the link.',
+            },
```
[the remark this answers is not in the document]
Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:378]: @mutebg  Shall I remove this?

File: site/pages/components/link.tsx (hunk 1 above)
Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:347]: Will check this @mutebg

File: site/pages/components/link.tsx (hunk 2 above)
Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:423]: @mutebg , Should we need to have separate tables for **inlineLink** and **standaloneLink**?

File: site/pages/components/link.tsx [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+      title: 'Link',
+      description:
+        'Links allow users to navigate to a new location or to additional information.',
+    }}
+    layoutProps={layoutProps}
+    pageIntroduction={{
+      type: 'Navigation',
+      name: 'Link',
+      hero: {
+        illustration: 'components/url-illustration',
```
[the remark this answers is not in the document]
Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:36]: will modify. -thanks!

File: site/pages/components/link.tsx [hunk 6]
```diff
@@ -0,0 +1,664 @@
...
+              attribute: 'logicalProps',
+              type: 'MQ<string>',
+              description: (
+                <>
+                  Logical props can define either padding or margins, depending
+                  on the element &#39;s writing mode, directionality, or text
+                  orientation.,
+                </>
+              ),
+            },
```
Comment (mutebg) [inline site/pages/components/link.tsx:460]: sorry but this is not correct, the logical props are the same as Fieldset for example.
Screenshot 2022-11-07 at 9 09 39

Comment (RashikaNewsUK) [reply site/pages/components/link.tsx]: Hi @mutebg ,
the changes has been done. Please have a look on it. -thanks!

File: site/pages/components/link.tsx [hunk 7]
```diff
@@ -0,0 +1,643 @@
...
+            ...(commonLogicalProps() as OverridesRowsProps[]),
+          ],
+          overridesSummary:
+            'The LinkInline has a range of predefined elements and attributes that can be overridden to define its appearance.',
+          propsFooter: (
+            <>
+              <InlineMessage
+                icon={infoIcon}
+                role="region"
+                aria-label="props link info"
```
Comment (mutebg) [inline site/pages/components/link.tsx:448]: You have an accessibility error, I think it comes from that InlineMessage and the fact you have 2 messages with the same aria-label.
```suggestion
                aria-label="props LinkInline info"
```

File: site/pages/components/link.tsx [hunk 8]
```diff
@@ -0,0 +1,643 @@
...
+                </>
+              ),
+            },
+          ],
+          propsFooter: (
+            <>
+              <InlineMessage
+                icon={infoIcon}
+                role="region"
+                aria-label="props link info"
```
Comment (mutebg) [inline site/pages/components/link.tsx:542]:
```suggestion
                aria-label="props LinkStandalone info"
```

File: site/pages/components/link.tsx [hunk 9]
```diff
@@ -0,0 +1,644 @@
...
+              attribute: 'logicalProps',
+              type: 'MQ<string>',
+              default: 'space010',
+              description: (
+                <>
+                  Logical props can define either padding or margins, depending
+                  on the element&apos;s writing mode, directionality, or text
+                  orientation.
+                </>
+              ),
```
Comment (mutebg) [inline site/pages/components/link.tsx:615]: here you need to do it the same way as the other component

**Review by mutebg — CHANGES_REQUESTED** [review]

File: site/pages/components/link.tsx (hunk 9 above)
Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:615]: Let me check and update @mutebg

Comment (RashikaNewsUK) [reply site/pages/components/link.tsx]: @mutebg  the changes has been done.Apologize for before review request. thanks!

**Review by mutebg — APPROVED** [review]

File: site/pages/components/link.tsx [hunk 10]
```diff
@@ -0,0 +1,638 @@
...
+            By default, there is no aria-label set in the link component. The
+            text inside the anchor will be read by the screen reader, so it is
+            important that links are descriptive. This is so that there is a
+            clear expectation about where the link will take the user. If the
+            text is not descriptive enough, an{' '}
+            <InlineCode>aria-label</InlineCode> will be necessary to understand
+            where the link is taking the user to.
+          </>
+        ),
+        tableRows: [
```
Comment (Xin00163) [inline site/pages/components/link.tsx:315]: Didn't see this in the spec. Is this needed?

File: site/pages/components/link.tsx [hunk 11]
```diff
@@ -0,0 +1,638 @@
...
+      introduction:
+        'Links allow users to navigate to a new location or to additional information.',
+    }}
+    componentDefaultsKey="link"
+    meta={{
+      status: MetaStatus.Supported,
+      introduced: 'v0.2.0',
+      codeUrl: 'https://github.com/newscorp-ghfb/newskit/tree/main/src/link',
+      figmaUrl:
+        'https://www.figma.com/file/FSbCQa6SzVR3K48ZWLeD77/%F0%9F%9F%A2-NK-Web-Components?node-id=324%3A4',
```
Comment (Xin00163) [inline site/pages/components/link.tsx:53]: I think now @mstuartf merged https://github.com/newscorp-ghfb/newskit/pull/449/. You will need to add storybookId like this [site/pages/components/button.tsx](https://github.com/newscorp-ghfb/newskit/pull/449/files#diff-455ba27a2d6bacd97fd3d2cebef9d286dce091e21453e55b6995ea3528240057R171)

Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:53]: Hi @Xin00163 ,
Thanks for your review.
Are you telling like this?
{
            title: 'Link',
            page: true,
            **id: 'site/pages/components/link.tsx',**
            description:
              'Links allow users to navigate to a new location or to additional information.',
            illustration: 'components/url-illustration',
          },

File: site/pages/components/link.tsx (hunk 10 above)
Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:315]: @Xin00163 this has been mentioned in the design link, so I have added. could you please refer this link FYR?  [https://www.figma.com/file/jj2dNnE0yZ9D1vYf14TFQe/02---NK-Web---Link?node-id=2034%3A315102&t=CiGvCXAt3EsPWouw-0](url)

File: site/pages/components/link.tsx (hunk 11 above)
Comment (RashikaNewsUK) [reply site/pages/components/link.tsx:54]: @Xin00163 added. Please have a look on it. thanks!

**Review by mutebg — APPROVED** [review]

**Review by Xin00163 — APPROVED** [review]

**Pull request merged**